weave.lisp

```diff
@@ -197,71 +227,110 @@
       (weave-uses weaver def))
   (write-line "</div>"))
 
+;; This is not elegant.
+(defun replace-sections-with-id-anchors (trie line)
+  (let ((r "")
+        (s 0)
+        (e 0))
+    (ppcre:do-scans (start end reg-starts reg-ends *anchor-pattern* line)
+      (let ((title (subseq line (aref reg-starts 0) (aref reg-ends 0))))
+        (setf
+         r
+         (format
+          nil
+          "~a~a"
+          r
+          (concatenate
+           'string
+           (subseq line s (- (aref reg-starts 0) 1))
+           (format nil "#~a" (cdr (assoc :terminal (cdr (trie-get trie (list title)))))))))
+        (setf s (aref reg-ends 0))
+        (setf e (aref reg-ends 0))))
+    (setf r (format nil "~a~a" r (subseq line e)))
+    r))
+
+(comment
+ (let ((trie (trie-put '() '("some-chapter") "c1")))
+   (replace-sections-with-id-anchors
+    trie
+    "foo [some chapter](@some-chapter) bar [other](@some-chapter) buz"))
+ ; => "foo [some chapter](#c1) bar [other](#c1) buz"
+ )
+
 (defun weave-prose-line (weaver line def)
   (loop for expr in line do
-        (cond ((stringp expr) (write-string expr)) 
```

Comment:
I didn't intend to make this blank space change. I have a habit of running `M-q` for `paredit-reindent-defun` and saved/committed this and didn't want to bother with undoing it yet.